Kernel/Devices/HID/USB/MouseDevice.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2022, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Badge.h>
+#include <AK/CircularQueue.h>
+#include <AK/OwnPtr.h>
+#include <AK/RefPtr.h>
+#include <Kernel/API/MousePacket.h>
+#include <Kernel/Bus/USB/USBDevice.h>
+#include <Kernel/Devices/HID/MouseDevice.h>
+#include <Kernel/Library/KString.h>
+#include <Kernel/Security/Random.h>
+
+namespace Kernel {
+
+class USBMouseDevice final : public MouseDevice {
```

Comment:
Should this inherit from `USBDevice` instead? See the USB MassStorage PR. Or we could argue it should inherit from both classes I guess. Not really sure what the pattern is throughout the code

Reply:
I dislike that pattern so I will not apply it here - what @Hendiadyoin1 did was essentially a hack with a singleton, due to a old limitation that we have with current code that any device should be controlled by a controller. What we really want in the future is to allow a device to be controlled directly be a controller or a `SOME_BUS::Driver` struct.